v0/destinations/revenue_cat/transform.js

```diff
@@ -0,0 +1,142 @@
+  defaultGetRequestConfig
+} = require("../../util");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  REVENUE_CAT_IDENTIFY_EXCLUSION
+} = require("./config");
+
+const trackResponseBuilder = async (message, category, { Config }) => {
+  if (!Config.xPlatform) {
+    throw new CustomError("[REVENUE CAT] X-Platform is required field.", 400);
+  }
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  // che
+  if (
+    message.properties.products &&
+    Array.isArray(message.properties.products)
+  ) {
+    const responseArray = [];
+    const productList = message.properties.products;
+
+    productList.forEach(product => {
+      const productDetails = constructPayload(
+        product,
+        MAPPING_CONFIG[CONFIG_CATEGORIES.PROPERTY.name]
+      );
+      payload = { ...payload, ...productDetails };
+      const response = defaultRequestConfig();
+      response.endpoint = "https://api.revenuecat.com/v1/receipts";
+
+      response.method = defaultPostRequestConfig.requestMethod;
+      const basicAuth = Buffer.from(Config.apiKey);
+      response.headers = {
+        Authorization: `Basic ${basicAuth}`,
+        "Content-Type": "application/json",
+        "X-Platform": `${Config.xPlatform}`
+      };
+      response.body.JSON = payload;
+      responseArray.push(response);
+    });
+    return responseArray;
+  }
+
+  const response = defaultRequestConfig();
+  response.endpoint = "https://api.revenuecat.com/v1/receipts";
+
+  response.method = defaultPostRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.apiKey);
+  response.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json",
+    "X-Platform": `${Config.xPlatform}`
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const identifyResponseBuilder = async (message, category, { Config }) => {
+  const returnValue = [];
+
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  let customPayload = {};
+  customPayload = extractCustomFields(
+    message,
+    customPayload,
+    ["traits", "context.traits"],
+    REVENUE_CAT_IDENTIFY_EXCLUSION
+  );
+
+  Object.entries(customPayload).forEach(([key, value]) => {
+    set(payload, `${key}.value`, value.toString());
+  });
+
+  const responseGet = defaultRequestConfig();
+  responseGet.endpoint = `https://api.revenuecat.com/v1/subscribers/${payload.app_user_id.value}`;
+  responseGet.method = defaultGetRequestConfig.requestMethod;
+  const basicAuth = Buffer.from(Config.apiKey);
+  responseGet.headers = {
+    Authorization: `Basic ${basicAuth}`,
+    "Content-Type": "application/json"
+  };
+  returnValue.push(responseGet);
+
+  const response = defaultRequestConfig();
+  response.endpoint = `https://api.revenuecat.com/v1/subscribers/${payload.app_user_id.value}/attributes`;
```

Comment:
Add constants for different endpoints in util and use them.

Reply:
Okay, done